mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+
+                // Take the next argument as the input file
+                input_fileGiven = true;
+                //std::cout << input_fileGiven << std::endl;
+                input_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an input file is " << input_file << std::endl;
+
+            }
+
+            
+        }
+        
+
+
+
+        // Looking for whether the option -o has been used as an argument
+        if(cmdLineArgs[i] == "-o" ){
+
+            // Need to look to see if this was the last argument specified
+            // If it was, need to return an error message.
+            if (i == (cmdLineArgs.size() - 1)){
+                std::cout << "An output file must be specified in the -o option after '-o'." << std::endl;
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+            
+                // Take the next argument as the output file
+                output_fileGiven = true;
+                output_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an output file is " << output_file << std::endl;
+            }
+           
+        }
+        
+
+        
+        
+    }
+
+
+    
+
+    // When help is requested
+    if (helpRequested) {
+        // Print help message
+        std::cout
+        << " \n"
+        << " mpags-cipher [-h/--help]\n" 
+        << " Encrypt and decrypt alphanumeric text using classical ciphers\n"
+        << " Avaliable options:\n"
+        << " -h | --help            Print this help message and exit\n"
+        << " --version              Print the version number and exit\n"
+        << " -i <input_file>        Specify the input file\n"
+        << " -o <output_file>       Specify the output file\n"
+        << " -c <cipher>            Specify the type of classical cipher:\n"
+        << "                                type1\n"
+        << "                                type2\n"
+        << "                                type3\n"
+        << " -k <key>               Specify the key\n"
+        << " encrypt/decrypt        Specify if the message is for encryption or decryption\n"
+        << std::endl;
+
+    
+    }
+
+    // When version is requested
+    if (versionRequested) {
+        // Print the version
+        std::cout
+        << " \n"
+        << " version: v0.1.0\n" 
+        << std::endl;
+
+        
+    }
+
+    // The input file 
+    if (input_fileGiven) {
+
+        // At the moment, this will just be a print statement
+        std::cout << "The input file to the program is " << input_file << " ." << std::endl;
+        
+    }
+    else{
+        std::cout << "The input file must be specified. Use -h / --help for more information." << std::endl;
+         
+
+    }
```

Comment:
There is no requirement that an input file be provided.  In such a case the input will be read from `std::cin`.

Similarly for the output, which should be written to `std::cout` if no output file is specified.